packages/next/src/server/dev/node-stack-frames.ts

```diff
@@ -25,6 +26,15 @@ function getFilesystemFrame(frame: StackFrame): StackFrame {
 }
 
 export function getServerError(error: Error, type: ErrorSourceType): Error {
+  const errMessage = error.message
+  const errorName = error.name
+  let errorStack = error.stack
+
+  // Retrieve the original error from the cause chain
```

Comment:
can you explain this part a bit? what happens if we don't do this post-processing?